### PR TITLE
feat(content): Added CheesyInnawoodFixes

### PR DESCRIPTION
**Pull request by cheesealmighty**
### Required

- [+ ] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [? ] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [? ] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [ ?] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

Okay, currently, innawoods content is limited to pre-industrial era, limiting player from accessing most of the content.
 This ports recipes that were made in the CDDA branch for the innawoods mod. Allows strip mining for obtaining most materials. 

## Describe the solution
Makes some recipes autolearn and adds few other recipes for already existing items. Also, adds a recipe for a makeshift transformer, which is basically copper wires wrapped around a steel bit. Also makes the gun recipes autolearned.

## Describe alternatives you've considered
None, currently.

## Testing
I've tested all the recipes in my world, it seems to be working fine.

### CODE REVIEW

**Review by RoyalFox2140 — CHANGES_REQUESTED** [review]
Need to add `"id_suffix"` first on duped recipes like bleach.

**Comment by RoyalFox2140** [comment]
Also we need to add a mod in semantics.

File: data/mods/CheesyInnaWoodsFixes/items/ammo_casing.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[
+  {
+    "id": "brass_metal",
+    "type": "GENERIC",
+    "category": "spare_parts",
+    "name": { "str_sp": "brass" },
+    "description": "Alloy of %70 copper and %30 zinc, useful for creating bullet casings.",
```
Comment (chaosvolt) [inline data/mods/CheesyInnaWoodsFixes/items/ammo_casing.json:7]:
```suggestion
    "description": "An alloy of 70% copper and 30% zinc, useful for creating bullet casings.",
```

**Review by RoyalFox2140 — APPROVED** [review]
Well its a little hard to review since it's so large but nothing stands out right now as needing fixing.

**Comment by Soadreqm** [comment]
I personally feel that most of those recipes _shouldn't_ be autolearned. What's even the point of looting libraries for advanced crafting books, if every forest-dwelling hobo just knows how to craft everything of interest?

**Review by scarf005 — COMMENTED** [review]
low priority concerns and questions:

- should it be a separate mod? wouldn't it be better to apply the changes to innawoods mod directly?
- please properly fill in checkboxes sections using [github flavoured markdown](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists).

**Comment by cheesealmighty** [comment]
> I personally feel that most of those recipes _shouldn't_ be autolearned. What's even the point of looting libraries for advanced crafting books, if every forest-dwelling hobo just knows how to craft everything of interest?

Well, yes, but there is a huge amount of content you have no access to in innawoods. The idea of this mod is to have a way to access those  content since no libraries spawn. And that's no fun.



> low priority concerns and questions:
> 
> * should it be a separate mod? wouldn't it be better to apply the changes innawoods mod  directly?
> * please properly fill in checkboxes sections using [github flavoured markdown](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists).

It probably shouldn't be a separate mod, but the idea was to test it out before merging it into the innawoods mod. I've been play testing in my own world, and I know it needs some tweaking and there are gaps. Like, you can craft car batteries but not storage batteries. I would appreciate the feedback in that regard.

File: data/mods/CheesyInnaWoodsFixes/recipes/new/smelting_bn.json [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    "book_learn": [ [ "textbook_armschina", 3 ], [ "textbook_fabrication", 3 ], [ "welding_book", 3 ] ],
+    "//": "Hematite has around 70% yield of iron, total yield here is just over 69% at 3500 grams of steel. Estimated 2% carbon content has minimal effect on these numbers. Magnetite has around 80% yield, but ironsand isn't pure magnetite.",
+    "result_mult": 75,
+    "using": [ [ "forging_standard", 4 ] ],
+    "tools": [ [ [ "crucible", -1 ], [ "crucible_clay", -1 ] ] ],
+    "components": [
+      [ [ "chunk_hematite", 1 ], [ "iron_ore", 1 ] ],
+      [ [ "material_shrd_limestone", 1 ], [ "material_limestone", 10 ] ],
+      [ [ "charcoal", 10 ], [ "coal_lump", 10 ] ]
+    ]
```
Comment (chaosvolt) [inline data/mods/CheesyInnaWoodsFixes/recipes/new/smelting_bn.json:176]: Good lord three entire recipes that do the same thing as a single vanilla recipe, plus using hematite ore that the mod authors of Innawood stole straight out of Mining Mod (gonna bet 90% odds they didn't credit them either).
```suggestion
```

**Review by chaosvolt — CHANGES_REQUESTED** [review]
Gwuh, an absolute ton of things that make no changes relative to vanilla or are otherwise redundant.

File: data/mods/CheesyInnaWoodsFixes/recipes/new/smelting_bn.json (hunk 2 above)
Comment (RoyalFox2140) [reply data/mods/CheesyInnaWoodsFixes/recipes/new/smelting_bn.json:176]: Remember to be nice :D

**Comment by RoyalFox2140** [comment]
As a mod I approved the recipe overwrites on the fair assumption they weren't autolearned or changed other stats like difficulty, I checked each overwrite but not to ensure they weren't fully identical.

**Comment by RoyalFox2140** [comment]
> I personally feel that most of those recipes _shouldn't_ be autolearned. What's even the point of looting libraries for advanced crafting books, if every forest-dwelling hobo just knows how to craft everything of interest?

So, something I want to do is more auto learn on recipes which is a factor on why I think skill gain is currently too fast, and why I also want to split science from cooking later. I think a lot of recipes bottlenecking you isn't very fun, example is glassblowing requires a specific book but has no use to the player unless they're a base builder.

**Review by chaosvolt — APPROVED** [review]
I...think it's good to go then. A lot of this is likely gonna need additional checking to ensure that all the added recipe overrides work as desired, in particular making sure that they're all identical with the vanilla recipe except for setting autolearn or swapping in relevant makeshift parts, but I think I got most of them.

**Review by RoyalFox2140 — APPROVED** [review]
Should be okay now. It's still hard to review something this large but we'll fix minor stuff in post.

**Pull request merged**